lib/fastlane/actions/update_url_schemes.rb

```diff
@@ -0,0 +1,68 @@
+require 'plist'
+
+module Fastlane
+  module Actions
+    class UpdateUrlSchemesAction < Action
+      def self.run(params)
+        path = params[:path]
+        url_schemes = params[:url_schemes]
+
+        hash = Plist.parse_xml(path)
+        hash['CFBundleURLTypes'].first['CFBundleURLSchemes'] = url_schemes
```

Comment:
The `defaults` command can take a file path, but if you check `man defaults`. You will see:

    WARNING: The defaults command will be changed in an upcoming
    major release to only operate on preferences domains. General
    plist manipulation utilities will be folded into a different
    command-line program.

So the best way is to manually parse the plist for now and that's what the `update_info_plist` action also does anyway.